pylint/testutils/output_line.py

```diff
@@ -16,19 +19,30 @@ class MessageTest(
 ):
     """Used to test messages produced by pylint. Class name cannot start with Test as pytest doesn't allow constructors in test classes."""
 
-    def __new__(cls, msg_id, line=None, node=None, args=None, confidence=None):
+    def __new__(
+        cls,
+        msg_id: str,
+        line: Optional[int] = None,
+        node: Optional[nodes.NodeNG] = None,
+        args: Any = None,
+        confidence: Optional[Confidence] = None,
+    ) -> "MessageTest":
         return tuple.__new__(cls, (msg_id, line, node, args, confidence))
 
-    def __eq__(self, other):
+    def __eq__(self, other: object) -> bool:
         if isinstance(other, MessageTest):
             if self.confidence and other.confidence:
                 return super().__eq__(other)
-            return self[:-1] == other[:-1]
+            return tuple(self[:-1]) == tuple(other[:-1])
```

Comment:
`mypy` complains that on this line the `self` to `__eq__` is no longer a `MessageTest`. Making both sides a tuple solves this issue.